src/ImageSharp/Formats/Jpeg/Components/Decoder/ColorConverters/JpegColorConverterVector.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Numerics;
+
+namespace SixLabors.ImageSharp.Formats.Jpeg.Components.Decoder.ColorConverters
+{
+    internal abstract partial class JpegColorConverterBase
+    {
+        /// <summary>
+        /// <see cref="JpegColorConverterBase"/> abstract base for implementations
+        /// based on <see cref="Vector"/> API.
+        /// </summary>
+        /// <remarks>
+        /// Converters of this family can work with data of any size.
+        /// Even though real life data is guaranteed to be of size
+        /// divisible by 8 newer SIMD instructions like AVX512 won't work with
+        /// such data out of the box. These converters have fallback code
+        /// for 'remainder' data.
+        /// </remarks>
+        internal abstract class VectorizedJpegColorConverter : JpegColorConverterBase
+        {
+            protected VectorizedJpegColorConverter(JpegColorSpace colorSpace, int precision)
+                : base(colorSpace, precision)
+            {
+            }
+
+            public sealed override bool IsAvailable => Vector.IsHardwareAccelerated && Vector<float>.Count % 4 == 0;
+
+            public override void ConvertToRgbInplace(in ComponentValues values)
+            {
+                DebugGuard.IsTrue(this.IsAvailable, $"{this.GetType().Name} converter is not supported on current hardware.");
+
+                int length = values.Component0.Length;
+                int remainder = length % Vector<float>.Count;
```

Comment:
```suggestion
                int remainder = (int)((uint)length % (uint)Vector<float>.Count);
```
results in [`& 7`](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABABgAJiBGAOgDkBXfGKASzFwG4BYAKD+IBmSgCZyAYXIBvPuTmVhrAHYZyAWSoAKZaoA2MJQHMMACwCU5ALwA+cvqOnyAUnIA1GGAzQAPADNdENgY1jTiEAwqPLzyCuQ66iLaKnYGxuZWtkkYZpqaDDpm9mnO5HkF7p4+/oHBoeEqZlEAvkA=) instead of the division.

The second cast is strange, as `Vector<float>.Count` should be treated as constant by the JIT.

Reply:
Filed https://github.com/dotnet/runtime/issues/62326 to address this in the JIT too.